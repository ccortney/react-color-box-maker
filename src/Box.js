import "./Box.css"

const Box = ({id, color, side, deleteBox}) => {
    return (
        <div className="Box">
            <div
                className="Box-div"
                style={{
                    backgroundColor: color,
                    height: `${side}px`,
                    width: `${side}px`
                }}
            >
            </div>
            <button className="Box-button" onClick={() => deleteBox(id)}>X</button>
        </div>

    )
}

export default Box;