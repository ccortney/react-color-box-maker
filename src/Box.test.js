import { render, screen } from '@testing-library/react';
import Box from './Box';

// smoke test
test('renders without crashing', () => {
  render(<Box />);
})

// snapshot test
test ('matches snapshot', () => {
  const {asFragment} = render(<Box/>);
  expect(asFragment()).toMatchSnapshot();
})
