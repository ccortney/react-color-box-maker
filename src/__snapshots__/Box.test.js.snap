// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      class="Box-div"
    />
    <button
      class="Box-button"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
